.travis.yml

```diff
@@ -16,34 +13,57 @@ before_script:
 script:
   - xvfb-run sbt -jvm-opts .jvmopts-travis "set concurrentRestrictions in Global += Tags.limit(ScalaJSTags.Link, 1)" ++$TRAVIS_SCALA_VERSION $TEST_PACKAGE
 
-jdk:
-  - openjdk11
+jdk: openjdk11
 
 scala:
   - 2.12.9
   - 2.11.12
 
 env:
-  - TEST_PACKAGE="guide-selenium/test"
-  - TEST_PACKAGE="udash-jvm/test"
-  - TEST_PACKAGE="utils-js/test"
-  - TEST_PACKAGE="core-js/test"
-  - TEST_PACKAGE="rpc-js/test"
-  - TEST_PACKAGE="rest-js/test"
-  - TEST_PACKAGE="i18n-js/test"
-  - TEST_PACKAGE="auth-js/test"
-  - TEST_PACKAGE="css-js/test"
-  - TEST_PACKAGE="bootstrap/test"
-  - TEST_PACKAGE="bootstrap4/test"
-  - TEST_PACKAGE="charts/compile"
-  - TEST_PACKAGE="benchmarks/compile"
-  - TEST_PACKAGE="doc"
+  matrix:
+    - TEST_PACKAGE="guide-selenium/test"
+    - TEST_PACKAGE="udash-jvm/test"
+    - TEST_PACKAGE="utils-js/test"
+    - TEST_PACKAGE="core-js/test"
+    - TEST_PACKAGE="rpc-js/test"
+    - TEST_PACKAGE="rest-js/test"
+    - TEST_PACKAGE="i18n-js/test"
+    - TEST_PACKAGE="auth-js/test"
+    - TEST_PACKAGE="css-js/test"
+    - TEST_PACKAGE="bootstrap/test"
+    - TEST_PACKAGE="bootstrap4/test"
+    - TEST_PACKAGE="charts/compile"
+    - TEST_PACKAGE="benchmarks/compile"
+    - TEST_PACKAGE="doc"
+
+  global:
+    - secure: fdBpnGod9OiHppoTk8v1uDnHcaG8vzZiORpqVLP0DWVvsdGxCfc2njUStxyD5LrNNTChd0W2aj9Ahchg88mlatheNvP/lnLpoU+1PIDNvshUmhOIVW1HDn4kkJQ783DW/Ch5fhRwupzr0LyW6BY+Ip+P2HIP3x/27ls8OUoX3Dr32BDPaRBQXl3RtFiic+pj1waB2qFK4Wbf4vWT0pSACaJXUGBWJq5LQ6QyTQ+pOcVBX65ZrsX5X6bSrgYapmkBgiNR/+cO7RIXeusHYmc6bchC8t6Rm/DfLosmR1iYMPGvVhTWaWo5t5x/xBAnEzBbKCHV7aE32OH69ed64/CcqtsgzPTT38Y5tvMCbDcUS3EJAqN1xEXdFcMWKkY4xzU2nekAIBdl9PT8Er2i/CCnMzG6/kxVCbnDD+Ujts+yJmWJPZ1dYdGkhhjqs9hpbden7UeAiRswVBt7fjB+crbpJxtLtWdRdwJnrbE7HnpTohG5/tfzZK98rSXou4KfigXmRneiObrp7n1OOOAiy/J70JxiT9hzMBEQVya2shx0v1TZgN39RrvYzN1fWdtvfrVke8XwQnjGxzgtOilQWqqUzBo+p7zx+Q+WALTPv/sU4//gmQFfIimf3b90KbcwzcCh3MwnTfANx0iScc9FktxfDozRvE3RTM/IHDKf9TJgaN8=
+    - secure: JfVPHMqEITjrDqHbVkQZCR4VLtLUFchunlea7+JoIWpQ2o0RuSUA7jUk+bBG+Avj6qwYSu+1/wmTQY/xRQBTEDTiuf8CKROIX0sPeyTvTcGm31V0z+Ajxu7x4xmLH2hr4/9y35oAwqy0knBK84J3VZ6xGRKO3JQAnk/iotpb2uGj6wjso2rQnG56NNVuM/WIZ5BwRtJrCP7rbyOJTMMT639+ECrmppsEZnODqgARhe5PUT+DRn5MXvH0sayt38SukUkahD5qcnMthP3jbNkaMrZ0RlIAzbOQUvtzt1rwHjeo71V2Dd6q3kLworbkmbmmCbxd7DtWAmTcMir7+91hXLP6r+3q2YFh5juqXXFEDjox0Nqt1nl6sfHAJMEOIuji7XoJLSAK5nOY5OsFZ5WK/yF7JlllaBUFGIJOeQbf4k70Nr6nmer7iHrvaD3H2jhtkyp5cET/0hMNpo7jSnvoAk5l0eZhleWbTgwzbHpjjw/yDPKS+S+q+2DmtTrHcDa1Z79YHps/9hlryGM3kT3mCr2cwFP+aoPnXKGdPuiY95LcVPHeI8TKIopU3lOMXEPX1fTY/xNDauwl9zZfKm2dwJU9Qa3uF5NSgr0s0nzUuj4+VJlTr6tLREAEznPESTwmm9qlHrJPX4aXEDCxRdfgSRFsGoRTqkvoiwozdv8+Z1c=
 
 matrix:
   exclude:
     - scala: 2.11.12
       env: TEST_PACKAGE="guide-selenium/test"
 
+jobs:
+  include:
+    - if: tag =~ ^v
```

Comment:
You really want to have separate jobs for different scala versions? This way you lose atomicity of this operation which may cause two problems:
* `sonatypeRelease` from one job may accidentally promote repository uploaded by another one
* you may end up with a partial release when one scala version succeeds but another one fails

Reply:
I don't, but a single job fails due to Travis time limits. I might have to do the release part manually then :(

Reply:
That sort of misses the entire point...

Reply:
I attempted:
- caching the targets - Scala usually doesn't care too much anyway, requires exactly same env variables for Travis to pick it up as well
- parallel upload to a staging repository (`sbt-sonatype` 3.1) - works nicely, but messy to encode and error-prone
- bundle uploading (`sbt-sonatype` 3.4) - this seems to be the best solution, it's fast enough to publish Scala 2.11 and 2.12 in 15 minutes without any crazy caching: https://travis-ci.org/UdashFramework/udash-core/jobs/581824091